### DISCUSSION TITLE
Pipelines : How to view the inputs and outputs at each step?

### DISCUSSION THREAD

**Discussion by efficienthacks** [microsoft/semantic-kernel#1844] [Q&A]
I'm trying to debug the [code ](https://github.com/MicrosoftDocs/semantic-kernel-docs/tree/main/samples/dotnet/03-Chaining-Functions) corresponding to [this documentation about Chaining Functions manually](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/chaining-functions?tabs=Csharp).

Specifically:

```cs
    var output = await _kernel.RunAsync(
        pipelineContext,
        GetNumbers,
        ExtractNumbersFromJson,
        MathFunction,
        CreateResponse);
```

I can set the breakpoints for the Native functions to see the inputs and outputs, but what about the Semantic functions?  I've opened up the `output` object and didn't see any properties that showed the steps at each stage.  How can this be done?

Thanks!

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
